openshift/entrypoint.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+# Dynamically add assigned UID/GID to /etc/passwd
+export USER_ID=$(id -u)
+export GROUP_ID=$(id -g)
+if [ `id -u` -ge 10000 ]; then
+    cat /etc/passwd | sed -e "s/^${USER_ID}:/builder:/" > /tmp/passwd
+    echo "postgres:x:${USER_ID}:${GROUP_ID}:PostgreSQL Server:${HOME}:/bin/bash" >> /tmp/passwd
+    cat /tmp/passwd > /etc/passwd
+    rm /tmp/passwd
+fi
+
+cat > /home/postgres/patroni.yml <<__EOF__
+bootstrap:
+  dcs:
+    postgresql:
+      use_pg_rewind: true
+  initdb:
+  - auth-host: md5
+  - auth-local: trust
+  - encoding: UTF8
+  - locale: en_US.UTF-8
+  - data-checksums
+  pg_hba:
+  - host all all 0.0.0.0/0 md5
+  - host replication ${PATRONI_REPLICATION_USERNAME} ${POD_IP}/16    md5
+restapi:
+  connect_address: '${POD_IP}:8008'
+postgresql:
+  connect_address: '${POD_IP}:5432'
+  authentication:
+    superuser:
+      password: '${PATRONI_SUPERUSER_PASSWORD}'
+    replication:
+      password: '${PATRONI_REPLICATION_PASSWORD}'
+  callbacks:
```

Comment:
We need this callback script only for the case when we run Patroni on kubernetes with Etcd, ZooKeeper or Consul.
The purpose of it - assign the label with the current role on the Pod and update endpoints of the leader (master) service by putting the ip of the leader (master) pod there.

Patroni already takes care about changing pod labels, and, as far as I remember on openshift we can't put ips from the pod network into the endpoint by security reason. Basically all above-mentioned facts make such callback script obsolete.
Originally I ended up on having this script because pod labels update was implemented at the last step and I needed something to test and play with.

Reply:
This was just copied from the kubernetes directory so I can't comment on your callbacks. The important part of the entrypoint.sh modification is on the /etc/passwd file:

```
export USER_ID=$(id -u)
export GROUP_ID=$(id -g)
if [ `id -u` -ge 10000 ]; then
    cat /etc/passwd | sed -e "s/^${USER_ID}:/builder:/" > /tmp/passwd
    echo "postgres:x:${USER_ID}:${GROUP_ID}:PostgreSQL Server:${HOME}:/bin/bash" >> /tmp/passwd
    cat /tmp/passwd > /etc/passwd
    rm /tmp/passwd
fi
```

Reply:
Regarding your other question, the UID is unique to the project, not just the single pod iteration/run. So each pod in the project has the same UID. In combination with the file system permissions set in the Dockerfile, the files will be accessible within the same project. I haven't tested moving the PV to a separate project, but I'm not sure I see that as a regular use case since that would likely require backup/restore.